PURL-TYPES.rst

```diff
@@ -87,6 +87,25 @@ bitnami
       pkg:bitnami/wordpress@6.2.0
       pkg:bitnami/wordpress@6.2.0?arch=arm64
 
+brew
+----
+``brew`` for Homebrew-based packages:
+
+- The default repository (tap) is ``homebrew/core``.
+
+  - The tap syntax is ``https://github.com/{org}/Homebrew-{tap}``, so
+    ``homebrew/core`` corresponds to the URL ``https://github.com/homebrew/homebrew-core``.
+- The ``name`` is the formula name.
+- The ``version`` is the formula version.
```

Comment:
**question (non-blocking):** When thinking about the examples below, I think you should clarify  how purl spec should handle version-in-formula formulae.

As in, if I want PostgreSQL 12.17 (current release as of this posting), should the purl be

    pkg:brew/postgresql@12@12.17

or 

    pkg:brew/postgresql@12.17

and let whatever's reading the purl — Homebrew, ostensibly — figure out how that version maps to our major-version formulae?

I speculate only the latter is a valid purl, a safe assumption because I don't think it's a good idea to allow two `@` in the purl.

Reply:
Thanks for calling this out -- my read of the purl spec is that the name and other components are always URL-escaped, e.g. `pkg:brew/postgresql@12@12.17` would actually be encoded as `pkg:brew/postgresql%4012@12.17` in a "wire format" context. So the purl *shoudl* always unambiguously map to a package, without the end client having to do additional disambiguation work.

This is a readability sacrifice, but my (potentially wrong) understanding is that purls are mostly meant to show up only in machine-readable contexts anyways.

Reply:
Sigh, it's not pretty, but if that's the way it needs to be, so be it. If that's the case, I'd recommend adding an example with that…

Reply:
Done! Looks like the other examples for other ecosystem purls also contain URL-escaping examples, so we're in-line here 🙂

Reply:
Yep, I added @colindean's example below. Adding an explicit callout also makes sense to me; I'll add some language for that.